chain/network/src/peer_manager/network_state/mod.rs

```diff
@@ -324,7 +329,7 @@ impl NetworkState {
     ) -> Result<(), RegisterPeerError> {
         let this = self.clone();
         let clock = clock.clone();
-        self.spawn(async move {
+        self.spawn("register_connection", async move {
```

Comment:
NetworkState async functions are weirdly written. Makes me wonder if there are better ways of doing this while keeping the constraints of cancellable/dropable futures that we want with NetworkState.